src/ctapipe/io/interpolation.py

```diff
@@ -0,0 +1,344 @@
+            if unit is None:
+                if self.expected_units[col] is not None:
+                    raise ValueError(
+                        f"{col} must have units compatible with '{self.expected_units[col].name}'"
+                    )
+            elif not self.expected_units[col].is_equivalent(unit):
+                if self.expected_units[col] is None:
+                    raise ValueError(f"{col} must have units compatible with 'None'")
+                else:
+                    raise ValueError(
+                        f"{col} must have units compatible with '{self.expected_units[col].name}'"
+                    )
+
+    def _check_interpolators(self, tel_id):
+        if tel_id not in self._interpolators:
+            if self.h5file is not None:
+                self._read_parameter_table(tel_id)  # might need to be removed
+            else:
+                raise KeyError(f"No table available for tel_id {tel_id}")
+
+    def _read_parameter_table(self, tel_id):
+        input_table = read_table(
+            self.h5file,
+            f"{self.telescope_data_group}/tel_{tel_id:03d}",
+        )
+        self.add_table(tel_id, input_table)
+
+
+class PointingInterpolator(Interpolator):
+    """
+    Interpolator for pointing and pointing correction data
+    """
+
+    telescope_data_group = "/dl0/monitoring/telescope/pointing"
+    required_columns = frozenset(["time", "azimuth", "altitude"])
+    expected_units = {"azimuth": u.rad, "altitude": u.rad}
+
+    def __call__(self, tel_id, time):
+        """
+        Interpolate alt/az for given time and tel_id.
+
+        Parameters
+        ----------
+        tel_id : int
+            telescope id
+        time : astropy.time.Time
+            time for which to interpolate the pointing
+
+        Returns
+        -------
+        altitude : astropy.units.Quantity[deg]
+            interpolated altitude angle
+        azimuth : astropy.units.Quantity[deg]
+            interpolated azimuth angle
+        """
+
+        self._check_interpolators(tel_id)
+
+        mjd = time.tai.mjd
+        az = u.Quantity(self._interpolators[tel_id]["az"](mjd), u.rad, copy=False)
+        alt = u.Quantity(self._interpolators[tel_id]["alt"](mjd), u.rad, copy=False)
+        return alt, az
+
+    def add_table(self, tel_id, input_table):
+        """
+        Add a table to this interpolator
+
+        Parameters
+        ----------
+        tel_id : int
+            Telescope id
+        input_table : astropy.table.Table
+            Table of pointing values, expected columns
+            are ``time`` as ``Time`` column, ``azimuth`` and ``altitude``
+            as quantity columns for pointing and pointing correction data.
+        """
+
+        self._check_tables(input_table)
+
+        if not isinstance(input_table["time"], Time):
+            raise TypeError("'time' column of pointing table must be astropy.time.Time")
+
+        input_table.sort("time")
```

Comment:
As we mentioned earlier, you should probably make a copy of the table here to avoid the unexpected side-effect that calling this function modifies the input table. So just change it to:
```python
input_table = input_table.copy().sort()
```
That may increase memory usage a small amount, but as the copy goes out of scope, I guess it should only retain the numpy arrays used in interp1d, so it shouldn't be a problem.